clients/drcachesim/common/trace_entry.h

```diff
@@ -191,6 +191,32 @@ typedef enum {
      */
     TRACE_MARKER_TYPE_CPU_ID,
 
+    /**
+     * The marker value contains the function id defined by user in droption
+     */
+    TRACE_MARKER_TYPE_FUNC_ID,
+
+    /**
+     * The marker value contains the return address of the function call,
+     * whose id is specified by the closest previous FUNC_ID marker entry
+     * XXX: replace return address with callstack information in the future
+     */
+    TRACE_MARKER_TYPE_FUNC_RETADDR,
+
+    /**
+     * The marker value contains one argument value of the function call,
+     * whose id is specified by the closest previous FUNC_ID marker entry.
+     * The number of such entries for one function invocation is expected to
+     * equal to the number of function arguments.
+     */
+    TRACE_MARKER_TYPE_FUNC_ARG,
+
+    /**
+     * The marker value contains return value of the function call,
+     * whose id is specified by the closest previous FUNC_ID marker entry
```

Comment:
`the return value`, and turn FUNC_ID into a link

Reply:
Done